### PR TITLE
Feature/v2 buttons

### PR DESCRIPTION
**Pull request by ph7jack**
#### Overview
- Refactor colors
- Refactor Sizes
- Add tests with 100% coverage

#### New Features
- Light Color
- Mini Button: squared, rounded, and circle
- Colors Customization
- Sizes Customization
- Ring Offset color customization
- Multicolor buttons

#### Breaking Changes
- Removed `$loadingDelay` prop, use the spinner attribute instead
- Removed `<x-circle-button />`, use the `<x-buttons.mini />` instead. The mini button can be **rounded** or **squared** 
- Renamed `<x-icon.spinner />` to `<x-wireui::icons.spinner />`
- Removed dark color

### CODE REVIEW

File: src/Support/Buttons/Colors/ColorPack.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace WireUi\Support\Buttons\Colors;
+
+abstract class ColorPack
+{
+    public function get(?string $color): Color
+    {
+        if (!$color || $color === 'default') {
```
Comment (joaopalopes24) [inline src/Support/Buttons/Colors/ColorPack.php:9]: Use ternary:

```
        return !$color || $color === 'default'
            ? $this->default()
            : data_get(
                target: $this->all(),
                key: $color,
                default: new Color($color),
            );
```

File: src/Support/Buttons/Sizes/SizePack.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
+<?php
+
+namespace WireUi\Support\Buttons\Sizes;
+
+abstract class SizePack
+{
+    public function get(?string $size): string
+    {
+        if (!$size || $size === 'default') {
```
Comment (joaopalopes24) [inline src/Support/Buttons/Sizes/SizePack.php:9]: Use ternary:

```
        return !$size || $size === 'default'
            ? $this->default()
            : data_get(
                target: $this->all(),
                key: $size,
                default: $size,
            );
```

File: src/View/Attribute.php [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    }
+
+    private function extractName(): ?string
+    {
+        return Str::of($this->directive)->before(':')->before('.');
+    }
+
+    private function extractValue(): ?string
+    {
+        if (!str_contains($this->directive, ':')) {
```
Comment (joaopalopes24) [inline src/View/Attribute.php:35]: Use ternary:

```
        return str_contains($this->directive, ':')
            ? Str::of($this->directive)->after(':')->before('.')
            : null;
```

File: src/View/Components/Buttons/Base.php [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+            $this->attributes->offsetSet('wire:loading.attr', 'disabled');
+            $this->attributes->offsetSet('wire:loading.class', '!cursor-wait');
+        }
+
+        return $this;
+    }
+
+    protected function tag(): string
+    {
+        if ($this->attributes->has('href')) {
```
Comment (joaopalopes24) [inline src/View/Components/Buttons/Base.php]: Use ternary:

`return $this->attributes->has('href') ? 'a' : 'button';`

File: src/config.php [hunk 5]
```diff
@@ -47,6 +48,55 @@
         'color'   => env('WIREUI_CARD_COLOR', 'bg-white dark:bg-secondary-800'),
     ],
 
+    /*
+        |--------------------------------------------------------------------------
```
Comment (joaopalopes24) [inline src/config.php]: Leave the comment inline as the Laravel default.

File: src/config.php [hunk 6]
```diff
@@ -47,6 +48,55 @@
...
+            'light'   => Colors\Light::class,
+        ],
+        'sizes' => [
+            'base' => Sizes\Base::class,
+            'icon' => Sizes\Icon::class,
+        ],
+    ],
+
+    /*
+        |--------------------------------------------------------------------------
```
Comment (joaopalopes24) [inline src/config.php]: Leave the comment inline as the Laravel default.

**Review by joaopalopes24 — CHANGES_REQUESTED** [review]

File: src/Support/Buttons/Colors/ColorPack.php (hunk 1 above)
Comment (ph7jack) [reply src/Support/Buttons/Colors/ColorPack.php:9]: Avoid complex ternary usages to keep the code simple to read.
Use ternary only when you have two simple values and no multiline

**Pull request merged**